services/prediction_service.py

```diff
@@ -1,25 +1,29 @@
+from venv import create
 from controllers import db
+from models.qa_model import QAModel
 from utils import first_runner
 from models import Prediction
+from services.create_read_update_service import CreateReadUpdateService
 
+class PredictionService(CreateReadUpdateService):
+    """ DataService, functions inherited from CreateReadUpdateService : read_by_id(id), create(entity_model), update(entity_model) """
 
-class PredictionService():
 
+    def __init__(self):
+        '''Constructor, take in the specific model class and pass the db.model back to the parent'''
+        super().__init__(Prediction)
+
+    #Wrapper function to create
     def create_prediction(self, prediction:Prediction)->Prediction:
-        prediction.prediction = first_runner.answer_tweet_question(prediction.datum.tweet, prediction.datum.question)
-        db.session.add(prediction)
-        db.session.commit()
-        saved_prediction = Prediction.query.filter(Prediction.id == prediction.id).first()
+        model_prediction = first_runner.answer_tweet_question(prediction.datum.tweet, prediction.datum.question)
+        prediction.prediction = model_prediction[0]
+         
+        saved_prediction = self.create(prediction)
         return saved_prediction
 
+    #Wrapper function to update
     def update_prediction(self, prediction: Prediction) -> Prediction:
-        prediction_retrieve = Prediction.query.filter(Prediction.id == prediction.id).first()
-        prediction_retrieve.alt_answer = prediction.alt_answer
-        prediction_retrieve.is_correct = prediction.is_correct
-        db.session.commit()
-        saved_prediction = Prediction.query.filter(Prediction.id == prediction.id).first()
+        # TODO: Logic to be added if required
```

Comment:
Any reason to override this method at this point?  Might be something we do in the future, but for now it seems like we just need the generic update?

Reply:
might need to update the DATA table

Reply:
That's fair, in any event, it's name should mirror that of the superclass so it overrides it.